src/main/java/mate/academy/springboot/datajpa/service/h2/ProductServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package mate.academy.springboot.datajpa.service.h2;
+
+import java.util.List;
+import java.util.Map;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.repository.specification.ProductSpecificationManager;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationManager;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+    private final SpecificationManager<Product> productSpecificationManager;
+
+    @Autowired
+    public ProductServiceImpl(
```

Comment:
it's not consistent, use constructor or @RequiredArgsConstructor, but not both of them